sound_byte/lib/model/user.dart

```diff
@@ -136,17 +163,23 @@ class User
     this._idToName.remove(id);
   }
 
-  // Takes a chat ID and adds it from the chats list
-  void addChat(String id)
+  // Takes a user ID and adds it to the chats list
+  void addChat(String uid)
   {
-    this.chats.add(id);
-    this._reference.updateData({'chats' : FieldValue.arrayUnion([id])});
+    Chat newChat = new Chat(uid);
+    this._reference.updateData({'chats' : FieldValue.arrayUnion([newChat.chatId])});
+    Firestore.instance.collection('Users')
+      .document(uid)
+      .updateData({'chats' : FieldValue.arrayUnion([newChat.chatId])});
+    this.chats.add(newChat);
```

Comment:
What is the second update Data for?

Reply:
The other user in the conversation.

This doesn't work that well anyway because of asynchronous functions, but it works a bit better than before (Right now it creates two collections, one with a Messages subcollection and the other with a participants field, and both users get a null chatID dropped into their chats fields).